src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -168,6 +175,19 @@ public override bool Execute()
                 // Summary
                 RestoreSummary.Log(log, restoreSummaries);
 
+                var projectsWithImportsChanged = new StringBuilder();
+                foreach (var summary in restoreSummaries)
+                {
+                    // Success does not matter here, as long as the files are not changed
+                    if(summary.BuildFilesChanged)
+                    {
+                        projectsWithImportsChanged.Append(summary.InputPath).Append(';');
```

Comment:
i think it shouldn't add an additional `;` at the end of the list. It will unnecessary increase load for msbuild to always ignore last item when they split on semicolon.